docs/in-depth/configuration-file.md

```diff
@@ -44,30 +45,51 @@ my_inputs = { temp_vector[1], temp_vector[3] }
 
-## Default inputs, outputs and parameters <a name=defaults></a> 
+A full list of modules available out-of-the-box, along with documentation about the modules' inputs, outputs and parameters, is available at MoMEMta's [technical documentation](https://momemta.github.io/MoMEMta/dev/group__modules.html).
 
-Some inputs and outputs are not defined by any module *per se* but are used by MoMEMta to pass and retrieve important quantities:
+## Declaring input particles
 
-Inputs:
-
-* `input::particles`: Vector of LorentzVectors passed as argument to `computeWeights()`
-
-* `input::met`: LorentzVector of the observed MET (optional argument to `computeWeights()`)
-
-* `cuba::ps_points`: Vector of doubles between 0 and 1. Phase-space point, ie. set of random numbers launched by Cuba to perform the integration. The length of the vector is the total number of integration dimensions declared by the modules. Instead of manually specifying the `cuba::ps_points/i` input tags, it is possible to use to `getpspoint()` function (which automatically increases “i” each time it is called).
-
-* `cuba::ps_weight`: (double) Weight associated with the phase-space point
-
-Outputs:
+Particles passed as input to the `computeWeights()` function (see [Calling MoMEMta](calling-momemta)) can be linked with the modules by registering them in the Lua configuration. For instance, for a particle declared in C++ as `momemta::Particle m_part1 { "part1", p1, 0 }`, one would do:
+```Lua
+local l_part1 = declare_input("part1") -- Pass as argument the name string of m_part1
+```
+The Lua object `l_part1` now contains the input tag needed to access the particle's 4-vector. For instance, defining a transfer function on the energy of `part1` can be done as:
+```Lua
+GaussianTransferFunctionOnEnergy.tf_part1 = {
+  ps_point = add_dimension(), -- A transfer function integrates over a variable (the particle's energy), so we need a new dimension in the integrated volume
+  reco_particle = l_part1.reco_p4, -- Pass the input tag corresponding to the experimentally reconstructed 4-vector of the particle, passed to 'computeWeights()'
+  sigma = 0.10 -- Take 10% resolution on the energy
+}
+```
+Since the transfer function integrates over the particle's energy, it generates new values for its energy. This defines a new, 'parton'-level 4-vector for `part1`, which can now be passed e.g. to a matrix element. In this example, the parton-level 4-vector can be accessed directly through the `tf_part1::output` input tag. It is also possible to register it with the `l_part1` object:
+```Lua
+l_part1.set_gen_p4("tf_part1::output")
+```
+When declaring further modules, instead of using `tf_part1::output`, it is now possible to use `l_part1.gen_p4` to access the parton-level 4-vector, i.e. these are equivalent:
+```Lua
+P4Printer.printer1 = { input = "tf_part1::output" }
+P4Printer.printer2 = { input = l_part1.gen_p4 }
+```
```

Comment:
This part is really nice, thanks, but think we should do things the way around: you first define inputs in the configuration file, and then pass them to the `computeWeights` function. It's mandatory to defines inputs, you cannot longer directly use hardcoded input tags. I'd also explicitly write that you access the reco input tag using the `reco_p4` variable: currently, it's only written in the lua snippet, not in the text.

Maybe it's worth specify also that `gen_p4` points by default to the same value that `reco_p4`: that's helpful if you don't need / want transfer function on the input, so that reco and gen quantities are the same thing.